pubnub-chat-impl/src/commonMain/kotlin/com/pubnub/chat/internal/ChatImpl.kt

```diff
@@ -796,8 +796,11 @@ class ChatImpl(
                     return@thenAsync emptyList<GetUnreadMessagesCounts>().asFuture()
                 }
                 val channels = memberships.map { membership -> membership.channel.id }
-                val channelsTimetoken =
-                    memberships.map { membership -> membership.lastReadMessageTimetoken ?: 0 }
+
+                // we need to increased returned lastReadMessageTimetoken by 1 because for odds numbers larger than 9007199254740991 server store them one bigger
```

Comment:
I think maybe I found the source of the problem:
in MembershipImpl.setLastReadMessageTimetoken(timetoken: Long):             
```
put(METADATA_LAST_READ_MESSAGE_TIMETOKEN, timetoken)
should be
put(METADATA_LAST_READ_MESSAGE_TIMETOKEN, timetoken.toString())
```

and in ChatImpl.markAllMessagesAsRead():
```
put(METADATA_LAST_READ_MESSAGE_TIMETOKEN, relevantLastMessageTimeToken)
should be
put(METADATA_LAST_READ_MESSAGE_TIMETOKEN, relevantLastMessageTimeToken.toString())
```

and then maybe (I don't know) we don't need the plus one

Reply:
Thanks. I will check it.